### ISSUE TITLE
Metricbeat and kubernetes events

### ISSUE THREAD

**Issue by evilezh** [elastic/beats#13192]
Metricbeat 7.3 runs within openshift 3.9 with following module configuration:
```yaml
- module: kubernetes
  metricsets:
  - state_node
  - state_deployment
  - state_replicaset
  - state_statefulset
  - state_pod
  - state_container
  period: 10s
  host: ${NODE_NAME}
  hosts: ["kube-state-metrics-1:8080"]
- module: kubernetes
  enabled: true
  metricsets:
    - event
- module: kubernetes
  enabled: true
  period: 10s
  metricsets:
  - apiserver
  hosts: ["https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}"]
  host: ${NODE_NAME}
  bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
  ssl.certificate_authorities:
    - /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt
```
Also I noticed logs like those are spamming at rapid rate .. 
![image](https://user-images.githubusercontent.com/5277173/62660420-e2d5ab00-b965-11e9-942a-df4cba2fb254.png)

and  here is all event rate from single metricbeat, excluding those repeating events:
![image](https://user-images.githubusercontent.com/5277173/62660631-80c97580-b966-11e9-9b60-3458f1cba1e7.png)
![image](https://user-images.githubusercontent.com/5277173/62660727-cf770f80-b966-11e9-92cb-9feb905d31c5.png)

and including ..
![image](https://user-images.githubusercontent.com/5277173/62660862-372d5a80-b967-11e9-9e3b-6232d7957ce1.png)

and here is two seperate runs of that deployment, graph shows kubernetes events. Run around 60-90 mins.
![image](https://user-images.githubusercontent.com/5277173/62660534-2f20eb00-b966-11e9-9e99-e407f38307fd.png)

Meanwhile it seems - I'm receiving other metrics, just events stop coming. I couldn't find any logs suggesting, that something is wrong.
On picture there is two different runs e.g. i killed pod then it runs again.

**Comment by jsoriano** [comment]
This is probably related to https://github.com/elastic/beats/issues/11917, that will hopefully get fixed soon with the help of https://github.com/elastic/beats/pull/13051

**Comment by exekias** [comment]
This should have been fixed with 7.5, as this is now handled by the kubernetes client
